app/controllers/courses_by_wiki_controller.rb

```diff
@@ -1,14 +1,15 @@
 # frozen_string_literal: true
 
 class CoursesByWikiController < ApplicationController
-  respond_to :html
+  respond_to :html, :json
 
   before_action :set_wiki
 
   def show
     @courses_list = Course.where(home_wiki: @wiki).order(id: :desc)
     @courses_list = @courses_list.where("year(created_at) = #{params[:year].to_i}") if params[:year]
     @presenter = CoursesPresenter.new(current_user: current_user, courses_list: @courses_list)
+    @courses = @presenter.courses
```

Comment:
As above, I think re-using the same action for both HTML and JSON will result in duplicate queries being run.